lib/index.d.ts

```diff
@@ -0,0 +1,43 @@
+import {Request} from 'express';
+import * as passport from 'passport';
+
+export interface StrategyOptions {
+  clientID: string;
+  clientSecret: string;
+  callbackURL: string;
+  passReqToCallback?: true;
+  scope?: string[];
+}
+
+export interface StrategyOptionsWithRequest {
+  clientID: string;
+  clientSecret: string;
+  callbackURL: string;
+  passReqToCallback: true;
+  scope?: string[];
+}
+
+export interface VerifyOptions {
+  message: string;
+}
+
+export interface VerifyCallback {
+  (error: any, user?: any, options?: VerifyOptions): void;
+}
+
+export interface VerifyFunctionWithRequest {
+  (req: Request, accessToken: string, refreshToken: string, profile: any, done: VerifyCallback): void;
+}
+
+export interface VerifyFunction {
+  (accessToken: string, refreshToken: string, profile: any, done: VerifyCallback): void;
+}
+
+declare class Strategy implements Strategy {
```

Comment:
Now we declare the strategy class 🎉 this is where it all ties together. It inherits from Passport's Startegy object so we tell it that so we can use any functions / values Passport attaches.

It's a bit confusing because we're defining a class with the same name as the one it implmenets, this is a qwirk of typescript that you can do this and it figures it out but we need to adher to the names so we don't get a choice.